source/extensions/filters/http/fault/fault_filter.cc

```diff
@@ -391,15 +401,29 @@ FaultFilterStats FaultFilterConfig::generateStats(const std::string& prefix, Sta
                                  POOL_GAUGE_PREFIX(scope, final_prefix))};
 }
 
-void FaultFilter::maybeIncActiveFaults() {
+bool FaultFilter::maybeDoFault() {
   // Only charge 1 active fault per filter in case we are injecting multiple faults.
+  // Since we count at most one active fault per filter, we also allow multiple faults
+  // per filter without checking for overflow.
   if (fault_active_) {
-    return;
+    return true;
+  }
+
+  // We only check for overflow when attempting to perform a fault. Note that this means that a
+  // single request might increment the counter more than once if it tries to apply multiple faults,
+  // though it is also possible for it to fail the first check then succeed on the second (should
+  // another thread decrement the fault).
+  // TODO(snowp): Is this behavior ideal? Should we track whether we've rejected faults for this
```

Comment:
this is the main open question in my mind, not sure how important we feel this is

Reply:
IMO it's probably fine either way. I would go with whatever is easiest and most clear to implement. Can you remove the TODO and make it clear what we have chosen? Also not sure if it's worth it to add docs on the stat that makes this mroe clear, up to you.

Reply:
I think the docs imply the behavior I changed it to - it counts the number of *faults* were skipped due to overflow - so I'll leave it at that. Updated the comment a bit and removed the TODO